num/__private/signed_integer_macros.h

```diff
@@ -0,0 +1,445 @@
+  }                                                               \
+  /** sus::concepts::ShrAssign trait. */                          \
+  inline void operator>>=(/* TODO: u32 */ uint32_t s)& noexcept { \
+    ::sus::check(s < T::BITS());                                  \
+    primitive_value >>= s;                                        \
+  }                                                               \
+  static_assert(true)
+
+#define _sus__signed_abs(T)                                                    \
+  /** Computes the absolute value of itself.                                   \
+   *                                                                           \
+   * The absolute value of ##T##::MIN() cannot be represented as an ##T##, and \
+   * attempting to calculate it will panic.                                    \
+   */                                                                          \
+  constexpr inline T abs() const& noexcept {                                   \
+    ::sus::check(primitive_value != MIN_PRIMITIVE);                            \
+    if (primitive_value >= 0)                                                  \
+      return primitive_value;                                                  \
+    else                                                                       \
+      return -primitive_value;                                                 \
+  }                                                                            \
+                                                                               \
+  /** Checked absolute value. Computes `abs()`, returning None if the current  \
+   * value is MIN().                                                           \
+   */                                                                          \
+  constexpr Option<T> checked_abs() const& noexcept {                          \
+    if (primitive_value != MIN_PRIMITIVE) [[likely]]                           \
+      return Option<T>::some(abs());                                           \
+    else                                                                       \
+      return Option<T>::none();                                                \
+  }                                                                            \
+                                                                               \
+  /** Saturating absolute value. Computes `abs()`, returning MAX if the        \
+   *  current value is MIN() instead of overflowing.                           \
+   */                                                                          \
+  constexpr T saturating_abs() const& noexcept {                               \
+    if (primitive_value != MIN_PRIMITIVE) [[likely]]                           \
+      return abs();                                                            \
+    else                                                                       \
```

Comment:
Nit: Could remove the `else`. Obviously doesn't matter tho.

Reply:
Yeah it's my Rust showing, which got me back into writing if/else to be a whole expression instead of early outs.